nav2_collision_monitor/src/polygon.cpp

```diff
@@ -168,51 +170,95 @@ bool Polygon::isShapeSet()
   return true;
 }
 
-void Polygon::updatePolygon()
+void Polygon::updatePolygon(const Velocity & cmd_vel_in)
 {
-  if (footprint_sub_ != nullptr) {
-    // Get latest robot footprint from footprint subscriber
-    std::vector<geometry_msgs::msg::Point> footprint_vec;
-    std_msgs::msg::Header footprint_header;
-    footprint_sub_->getFootprintInRobotFrame(footprint_vec, footprint_header);
-
-    std::size_t new_size = footprint_vec.size();
-    poly_.resize(new_size);
-    polygon_.header.frame_id = base_frame_id_;
-    polygon_.polygon.points.resize(new_size);
-
-    geometry_msgs::msg::Point32 p_s;
-    for (std::size_t i = 0; i < new_size; i++) {
-      poly_[i] = {footprint_vec[i].x, footprint_vec[i].y};
-      p_s.x = footprint_vec[i].x;
-      p_s.y = footprint_vec[i].y;
-      polygon_.polygon.points[i] = p_s;
-    }
-  } else if (!polygon_.header.frame_id.empty() && polygon_.header.frame_id != base_frame_id_) {
-    // Polygon is published in another frame: correct poly_ vertices to the latest frame state
-    std::size_t new_size = polygon_.polygon.points.size();
-
-    // Get the transform from PolygonStamped frame to base_frame_id_
-    tf2::Transform tf_transform;
-    if (
-      !nav2_util::getTransform(
-        polygon_.header.frame_id, base_frame_id_,
-        transform_tolerance_, tf_buffer_, tf_transform))
-    {
-      return;
-    }
-
-    // Correct main poly_ vertices
-    poly_.resize(new_size);
-    for (std::size_t i = 0; i < new_size; i++) {
-      // Transform point coordinates from PolygonStamped frame -> to base frame
-      tf2::Vector3 p_v3_s(polygon_.polygon.points[i].x, polygon_.polygon.points[i].y, 0.0);
-      tf2::Vector3 p_v3_b = tf_transform * p_v3_s;
+  switch (polygon_source) {
+    case STATIC_POINTS:
+      {
+        if (!polygon_.header.frame_id.empty() && polygon_.header.frame_id != base_frame_id_) {
+          // Polygon is published in another frame: correct poly_ vertices to the latest frame state
+          std::size_t new_size = polygon_.polygon.points.size();
+
+          // Get the transform from PolygonStamped frame to base_frame_id_
+          tf2::Transform tf_transform;
```

Comment:
Breakout into new method `transformPolygon(polygon)`

Reply:
Also I'd think that a bunch of this can be made into a util and shared between mutliple instance where this is done

Reply:
not relevant anymore in the latest changes.